src/Arcus.WebApi.Unit/Correlation/TelemetryCorrelationTests.cs

```diff
@@ -0,0 +1,114 @@
+                request.Headers.Add("X-Transaction-ID", firstCorrelationInfo.TransactionId);
+                
+                using (HttpResponseMessage secondResponse = await client.SendAsync(request))
+                {
+                    Assert.Equal(HttpStatusCode.OK, secondResponse.StatusCode);
+                    CorrelationInfo secondCorrelationInfo = await AssertAppCorrelationInfoAsync(secondResponse);
+                    AssertLoggedCorrelationProperties(secondCorrelationInfo);
+
+                    Assert.NotEqual(firstCorrelationInfo.OperationId, secondCorrelationInfo.OperationId);
+                    Assert.Equal(firstCorrelationInfo.TransactionId, secondCorrelationInfo.TransactionId);
+                }
+            }
+        }
+
+        [Fact]
+        public async Task SendRequest_WithSerilogCorrelationenrichment_ReturnsOkWithDifferentOperationIdAndDifferentTransactionId()
+        {
+            // Arrange
+            using (HttpClient client = _testServer.CreateClient())
+            // Act
+            using (HttpResponseMessage firstResponse = await client.GetAsync(Route))
+            {
+                // Assert
+                Assert.Equal(HttpStatusCode.OK, firstResponse.StatusCode);
+                CorrelationInfo firstCorrelationInfo = await AssertAppCorrelationInfoAsync(firstResponse);
+                AssertLoggedCorrelationProperties(firstCorrelationInfo);
+
+                using (HttpResponseMessage secondResponse = await client.GetAsync(Route))
+                {
+                    Assert.Equal(HttpStatusCode.OK, secondResponse.StatusCode);
+                    CorrelationInfo secondCorrelationInfo = await AssertAppCorrelationInfoAsync(secondResponse);
+                    AssertLoggedCorrelationProperties(secondCorrelationInfo);
+                    
+                    Assert.NotEqual(firstCorrelationInfo.OperationId, secondCorrelationInfo.OperationId);
+                    Assert.NotEqual(firstCorrelationInfo.TransactionId, secondCorrelationInfo.TransactionId);
+                }
+            }
+        }
+
+        private static async Task<CorrelationInfo> AssertAppCorrelationInfoAsync(HttpResponseMessage response)
+        {
+            string json = await response.Content.ReadAsStringAsync();
+            var content = JsonConvert.DeserializeAnonymousType(json, new { TransactionId = "", OperationId = "" });
+            Assert.False(String.IsNullOrWhiteSpace(content.TransactionId), "Accessed 'X-Transaction-ID' cannot be blank");
+            Assert.False(String.IsNullOrWhiteSpace(content.OperationId), "Accessed 'X-Operation-ID' cannot be blank");
+
+            return new CorrelationInfo(content.OperationId, content.TransactionId);
+        }
+
+        private void AssertLoggedCorrelationProperties(CorrelationInfo correlationInfo)
+        {
+            IEnumerable<KeyValuePair<string, LogEventPropertyValue>> properties = 
```

Comment:
This is tricky, if we do a 2nd request it will still contain the traces from the first call, so the test can be successful while it's wrong, no?

Reply:
Yes, that's why I called the method `Dequeue...` because it removes the log events in the in-memory sink after it's called.

Reply:
Ah my bad, sorry - Looks like I'm having a bad day :/